### PR TITLE
detect_cycle added

### PR DESCRIPTION
**Pull request by jujusudhur**
**Fixes issue:** #2921 
**Changes:**
Added linked list detect cycle in python

### CODE REVIEW

File: code/data_structures/linked_list/circular_linked_list/has_loop.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
+import collections
+ +
```
Comment (arnavb) [inline code/data_structures/linked_list/circular_linked_list/has_loop.py]: Please remove all these + signs!

**Comment by arnavb** [comment]
Open a topic branch for separate changes.

**Comment by jujusudhur** [comment]
- Reset the previous commit, with changes pertaining to only this pull request. 
- Will create a separate branch for bead_sort.

**Comment by AdiChat** [comment]
Looks great 😍 

Hope you are enjoying your journey 🚆  with **OpenGenus**

**Pull request merged**